test/Mono.Linker.Tests.Cases/Reflection/EventsUsedViaReflection.cs

```diff
@@ -0,0 +1,254 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+
+namespace Mono.Linker.Tests.Cases.Reflection
+{
+	[SetupCSharpCompilerToUse ("csc")]
+	[ExpectedNoWarnings]
+	[SetupLinkerArgument ("--disable-opt", "unreachablebodies")]
+	public class EventsUsedViaReflection
+	{
+		public static void Main ()
+		{
+			new Foo (); // Needed to avoid lazy body marking stubbing
+
+			TestGetEvents ();
+			TestInternal ();
+			TestBindingFlags ();
+			TestUnknownBindingFlags (BindingFlags.Public);
+			TestNullType ();
+			TestDataFlowType ();
+			TestDataFlowWithAnnotation (typeof (MyType));
+			TestIfElse (1);
+			TestIgnoreCaseBindingFlags ();
+			TestUnsupportedBindingFlags ();
+		}
+
+		[Kept]
+		[RecognizedReflectionAccessPattern]
+		static void TestGetEvents ()
+		{
+			var events = typeof (Foo).GetEvents ();
+		}
+
+		[Kept]
+		// The event will not be kept as it's internal and the behavior of Type.GetEvent(string) is to only return public events
+		// But we don't mark it as unrecognized access pattern - we did recognize it fully, just didn't find the event being asked for
+		// The behavior of the code will not change by linking it:
+		//   - Without linking the GetEvent will return null
+		//   - After linking the GetEvent will still return null
+		// We also don't mark it as recognized pattern since we didn't mark anything
```

Comment:
```suggestion
		//   - Without linking the GetEvents will return null
		//   - After linking the GetEvents will still return null
		// We also don't mark it as recognized pattern since we didn't mark anything
```